catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/SearchFormsPredicatesIT.java

```diff
@@ -0,0 +1,367 @@
+    testJsonToXml(
+        getJsonFilter("=", "language", "english"),
+        getXmlFilter("PropertyIsEqualTo", "language", "english"));
+  }
+
+  @Test
+  public void testTextEqualXmlToJson()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    testXmlToJson(
+        getXmlFilter("PropertyIsEqualTo", "language", "english"),
+        getJsonResponse("=", "language", "english"));
+  }
+
+  @Test
+  public void testDateBetweenJsonToXml() throws IngestException, SourceUnavailableException {
+    testJsonToXml(
+        getJsonFilter("DURING", "created", CANNED_ISO_DATE + "/" + CANNED_ISO_DATE),
+        getXmlFilter("During", "created", CANNED_EPOCH_DATE + "/" + CANNED_EPOCH_DATE));
+  }
+
+  @Test
+  public void testDateBetweenXmlToJson()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    testXmlToJson(
+        getXmlFilter("During", "created", CANNED_EPOCH_DATE + "/" + CANNED_EPOCH_DATE),
+        getJsonResponse(
+            "DURING",
+            "created",
+            CANNED_ISO_DATE + "/" + CANNED_ISO_DATE,
+            CANNED_ISO_DATE,
+            CANNED_ISO_DATE));
+  }
+
+  @Test
+  public void testDateEmptyJsonToXml() throws IngestException, SourceUnavailableException {
+    testJsonToXml(
+        // counteract whitespace removal
+        getJsonFilter("IS NULL", "created", null).replace("ISNULL", "IS NULL"),
+        getNilXmlFilter("created"));
+  }
+
+  @Test
+  public void testDateEmptyXmlToJson()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    testXmlToJson(
+        getNilXmlFilter("created"),
+        // counteract whitespace removal
+        getJsonResponse("IS NULL", "created").replace("ISNULL", "IS NULL"));
+  }
+
+  private static void testJsonToXml(String requestJson, String expectedXml)
+      throws IngestException, SourceUnavailableException {
+    ArgumentCaptor<CreateRequest> requestCaptor = ArgumentCaptor.forClass(CreateRequest.class);
+
+    // Prepare
+    MetacardImpl metacardWithIdAndCreatedDate = new MetacardImpl();
+    metacardWithIdAndCreatedDate.setId(CANNED_ID);
+    metacardWithIdAndCreatedDate.setCreatedDate(new Date());
+
+    doReturn(
+            new CreateResponseImpl(
+                new CreateRequestImpl(Collections.emptyList()),
+                Collections.emptyMap(),
+                Collections.singletonList(metacardWithIdAndCreatedDate)))
+        .when(MOCK_FRAMEWORK)
+        .create(requestCaptor.capture());
+
+    // Execute
+    int statusCode =
+        RestAssured.given()
+            .header(CONTENT_IS_JSON)
+            .content(requestJson)
+            .post(localhostFormsUrl)
+            .statusCode();
+
+    Metacard searchForm = requestCaptor.getValue().getMetacards().get(0);
+    String capturedXml = ((QueryTemplateMetacard) searchForm).getFormsFilter();
```

Comment:
any chance that this cast will fail?

Reply:
No the only metacard impl that has a search form XML blob on it is the query template one.